### PR TITLE
Establish a staging repository for mirroring Docker Hub images

### PR DESCRIPTION
**Pull request by justaugustus**
Primary owners: @kubernetes/release-managers under the @kubernetes/release-engineering 
                subproject of SIG Release.

Shared ownership across:
- @kubernetes/sig-release 
- @kubernetes/sig-testing 
- @kubernetes/k8s-infra-team 

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

/assign @thockin @dims @BenTheElder @saschagrunert 
cc: @kubernetes/sig-release-leads 

ref: https://kubernetes.slack.com/archives/CCK68P2Q2/p1605906060178100, https://github.com/kubernetes/kubernetes/pull/95567, https://github.com/kubernetes/test-infra/issues/19477, https://www.docker.com/pricing/resource-consumption-updates, https://cloud.google.com/container-registry/docs/pulling-cached-images

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: groups/sig-release/groups.yaml
Comment (BenTheElder) [inline groups/sig-release/groups.yaml:165]: baby nit: I'm not in that group anymore (I am in google build admins tho, different rotation)
bigger nit: clarketm is not and is not at google anymore.

**Comment by justaugustus** [comment]
/hold cancel

Dropped test-infra-oncall in https://github.com/kubernetes/k8s.io/pull/1441/commits/754294995f7b7554045fbe8d48d6b4e053a03045 and opted for @kubernetes/build-admins instead. (ref: https://github.com/kubernetes/k8s.io/pull/1441#discussion_r528014985)

Needs re-LGTM.

**Pull request merged**
